sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedProcessorBuilderImpl.java

```diff
@@ -209,6 +214,92 @@ public boolean isStarted() {
+    public Mono<List<ChangeFeedProcessorState>> getCurrentState() {
+        List<ChangeFeedProcessorState> earlyResult = new ArrayList<>();
+
+        if (this.leaseStoreManager == null || this.feedContextClient == null) {
+            return Mono.just(earlyResult);
+        }
+
+        return this.leaseStoreManager.getAllLeases()
+            .flatMap(lease -> {
+                ChangeFeedOptions options = new ChangeFeedOptions()
+                    .setMaxItemCount(1)
+                    .setPartitionKeyRangeId(lease.getLeaseToken())
+                    .setStartFromBeginning(true)
+                    .setRequestContinuation(lease.getContinuationToken());
+
+                return this.feedContextClient.createDocumentChangeFeedQuery(this.feedContextClient.getContainerClient(), options)
+                    .take(1)
+                    .map(feedResponse -> {
+                        final String pkRangeIdSeparator = ":";
+                        final String segmentSeparator = "#";
+                        final String lsnPropertyName = "_lsn";
+                        final String tsPropertyName = "_ts";
+
+                        String sessionTokenLsn = feedResponse.getSessionToken();
+                        String parsedSessionToken = sessionTokenLsn.substring(sessionTokenLsn.indexOf(pkRangeIdSeparator));
+                        String[] segments = parsedSessionToken.split(segmentSeparator);
+                        String latestLsn = segments[0];
+
+                        if (segments.length >= 2) {
+                            // default to Global LSN
+                            latestLsn = segments[1];
+                        }
+
+                        ChangeFeedProcessorState changeFeedProcessorState = new ChangeFeedProcessorState()
+                            .setId(lease.getId())
+                            .setHostName(lease.getOwner())
+                            .setLeaseToken(lease.getLeaseToken())
+                            .setLastUpdate(ZonedDateTime.parse(lease.getTimestamp()).toInstant());
+
+
+                        // An empty list of documents returned means that we are current (zero lag)
+                        if (feedResponse.getResults() == null || feedResponse.getResults().size() == 0) {
+                            changeFeedProcessorState.setEstimatedLag(0);
+                            changeFeedProcessorState.setContinuationToken(latestLsn);
+
+                            return changeFeedProcessorState;
+                        }
+
+                        changeFeedProcessorState.setContinuationToken(feedResponse.getResults().get(0).get(lsnPropertyName).asText(null));
+
+                        try {
+                            changeFeedProcessorState.setContinuationTokenTimestamp(Instant.ofEpochMilli(Long.valueOf(
+                                    feedResponse.getResults().get(0).get(tsPropertyName).asText("0"))));
+                        } catch (NumberFormatException ex) {
```

Comment:
Wouldn't this indicate a critical failure - tracing as warning and ignoring unexpected backend response seems to make it harder than necessary to identify breaking changes / debug it?

Reply:
_ts should always be a long - right ?

Reply:
The error though unexpected (_ts system property is an epoch), it is not fatal (there's a log warning capturing this). Setting the continuation token as "null" is an indicator that the document found is not valid; i.e. we initialized CFP leases but we are yet to process any changes in this particular scope/partition.